Move data storage to PostgreSQL / SQLAlchemy

catherinedevlin opened the pull request:
Ready for review and merge:
- Store data in PostgreSQL
- Fetch as many issues as available (not just 100)
- Restructuring, moved logic into models

Test deployment visible at https://catherine-sitedata.18f.gov

catherinedevlin committed:
Store fetch results in sql

catherinedevlin committed:
Fix bug in saving

catherinedevlin committed:
Replace missing app.py changes, moved more logic to model

catherinedevlin committed:
Handle empty results for authors; include labels with issues

catherinedevlin committed:
Move `utils.py`.  Delete unused code.

catherinedevlin committed:
Trigger travis to build database

catherinedevlin committed:
Refresh updated issues.  Account for pagination in github API.

catherinedevlin committed:
Record Events for possible future analysis

Sorry to rename a migration, that's not how it's supposed to work.

catherinedevlin committed:
Move data-related functionality into model.py

catherinedevlin committed:
Move database backend to postgresql

Commit:
Environment expects PROD and STAGING env vars

Commit:
non-empty strings for PROD, STAGING

Commit:
use a default `since` date in load_issues

Commit:
Moved constant definition

Commit:
Store created & updated in database as dates, not datetimes

Datetimes were confusing app/static/js/app.js

catherinedevlin committed:
Update tests to match new `fetch_issues` workings

anthonygarvan commented:
As discussed, please review tests.py and update anything that assumes a database-less system.

catherinedevlin committed:
Automated 2to3 changes

catherinedevlin committed:
Test adjustments for db-based data storage

Removed unused tests
Removed unneeded Fetch class
Added fetch_issues tests
Changed syntax for Py3 compatibility

catherinedevlin committed:
Improved default and documentation for `manage.py updatedata`

catherinedevlin committed:
Merge branch 'master' into sql

catherinedevlin committed:
Folded `usage.txt` into `README.md` and `.env-example`

catherinedevlin committed:
Applied YAPF formatting

catherinedevlin committed:
Fixed tests for Py2/Py3 compatibility

Deleted `GitHub.get_repo_contents`; it was unused, and hardcoded
to use `contents` attribute of response, which specifies bytes
form of response & causes Py2/Py3 inconsistency.

catherinedevlin committed:
Merge branch 'master' into sql

Conflicts:
	app/app.py

Quite a bit of conflict was resolved was in code that had been
removed from the SQL branch.

catherinedevlin committed:
Pull duty station for authors from hub's biographies

catherinedevlin committed:
Get team information from 18F site API & store

catherinedevlin committed:
Cleanup comments on models for airport code, team

catherinedevlin committed:
Setup database upon deploy

catherinedevlin committed:
Merge branch 'master' into sql

Conflicts:
	.gitignore

catherinedevlin committed:
Handle empty/irrelevant environment variables better
